TypeCobol/Compiler/Symbols/SymbolExpander.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TypeCobol.Compiler.Types;
+using Type = System.Type;
+
+namespace TypeCobol.Compiler.Symbols
+{
+    /// <summary>
+    /// Symbol Expander
+    /// </summary>
+    public class SymbolExpander : Symbol.AbstractSymbolVisitor<Symbol, Symbol>
```

Comment:
It looks like SymbolExpander/TypedefExpander, SymbolValidator/TypeValidator and LevelRenumber/TypeLevelRenumber are similar. They are co-dependent and I think it would be more robust to merge them as one class extending AbstractSymbolAndTypeVisitor (i.e. Expander, Validator, LevelRenumber)

Reply:
This is because one can expand a Symbol and another one a Type (TypeDef), TypeDef can contains Symbols To expand. SymbolExpander can be given a specific TypeExpander instance to be used and TypeExpander can be given a specific SymbolExpander instance to be used.

Reply:
Yes we may end up with several instances of either SymbolExpander and TypeExpander for the same program but all these instances would behave the same. So I think it's cleaner to have a single class handling Symbol and Type expansion. Same thing for Validator and LevelRenumber.
As a side note, the property ShallLevelRenumber in SymbolExpander class is written only but never read, it should be removed.

Reply:
So there is no need to rewrite things here for the moment, it works like that, there is no duplication, you want unification, but it's ok for that moment.

Reply:
I keep on wanted to be able to give a specialized SymboExpander or TypeExpander, then the default ones.

Reply:
Ok, there is one thing that must be changed, because default constructors automatically create a corresponding SymbolExpander or TypeExpander if one is not supplied. This lead to have an unsused instance which will be Garbage Collected. Ok fm-117 is right by suggesting to provide for instance an ExpanderInstanceFactory, to avoid creating unused instances.